pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -457,6 +457,19 @@ func UpdateConfigSpecDeviceGroups(
 	}
 }
 
+// UpdateConfigSpecHardwareVersion sets the hardware version in the ConfigSpec
+// if the current hardware version is lower than the specified minimum version.
+func UpdateConfigSpecHardwareVersion(
+	config *vimTypes.VirtualMachineConfigInfo,
+	configSpec *vimTypes.VirtualMachineConfigSpec,
+	minHardwareVersion int32) {
+
+	existingVersion := util.ParseVirtualHardwareVersion(config.Version)
+	if existingVersion < minHardwareVersion {
+		util.EnsureMinHardwareVersionInConfigSpec(configSpec, minHardwareVersion)
```

Comment:
Maybe just `configSpec.Version = fmt.Sprintf()` here instead, since the `EnsureMinHardwareVersionInConfigSpec` kind of ends up doing the same check, and to make it clearer here that the field is only set if really needs to be changed so we don't otherwise Reconfigure.